src/analysis/normed_space/affine_isometry.lean

```diff
@@ -587,3 +588,32 @@ begin
   rw this,
   simp only [homeomorph.comp_is_open_map_iff, homeomorph.comp_is_open_map_iff'],
 end
+
+local attribute [instance, nolint fails_quickly] affine_subspace.nonempty_map
+
+include V₁
+omit V
+
+/-- Restricts an affine isometry to an affine isometry equivalence between a nonempty affine
+subspace `E` and its image. -/
+noncomputable def affine_isometry.restrict_to_equiv
+  (E : affine_subspace 𝕜 P₁) [nonempty E]
+  (φ : P₁ →ᵃⁱ[𝕜] P₂) : E ≃ᵃⁱ[𝕜] E.map φ.to_affine_map :=
```

Comment:
@YaelDillies, do you know if we have something like this already for other subobjects? It feels a bit like your #17775 PR.